lib/lte_link_control/lte_lc.c

```diff
@@ -630,6 +957,262 @@ static int parse_nw_reg_status(const char *at_response,
 	return err;
 }
 
+/**@brief Parses an AT command response, and returns the current RRC mode.
+ *
+ * @param at_response Pointer to buffer with AT response.
+ * @param mode Pointer to where the RRC mode is stored.
+ * @param mode_index Parameter index for mode.
+ *
+ * @return Zero on success or (negative) error code otherwise.
+ */
+static int parse_rrc_mode(const char *at_response,
+			  enum lte_lc_rrc_mode *mode,
+			  size_t mode_index)
+{
+	int err, temp_mode;
+	struct at_param_list resp_list = {0};
+
+	if ((at_response == NULL) || (mode == NULL)) {
+		return -EINVAL;
+	}
+
+	err = at_params_list_init(&resp_list, AT_CSCON_PARAMS_COUNT_MAX);
+	if (err) {
+		LOG_ERR("Could not init AT params list, error: %d", err);
+		return err;
+	}
+
+	/* Parse CSCON response and populate AT parameter list */
+	err = at_parser_max_params_from_str(at_response,
+					    NULL,
+					    &resp_list,
+					    AT_CSCON_PARAMS_COUNT_MAX);
+	if (err) {
+		LOG_ERR("Could not parse +CSCON response, error: %d", err);
+		goto clean_exit;
+	}
+
+	/* Get the RRC mode from the response */
+	err = at_params_int_get(&resp_list, mode_index, &temp_mode);
+	if (err) {
+		LOG_ERR("Could not get signalling mode, error: %d", err);
+		goto clean_exit;
+	}
+
+	/* Check if the parsed value maps to a valid registration status */
+	if (temp_mode == 0) {
+		*mode = LTE_LC_RRC_MODE_IDLE;
+	} else if (temp_mode == 1) {
+		*mode = LTE_LC_RRC_MODE_CONNECTED;
+	} else {
+		LOG_ERR("Invalid signalling mode: %d", temp_mode);
+	}
+
+clean_exit:
+	at_params_list_free(&resp_list);
+
+	return err;
+}
+
+/* Confirm valid system mode and set Paging Time Window multiplier.
+ * Multiplier is 1.28 s for LTE-M, and 2.56 s for NB-IoT, derived from
+ * Figure 10.5.5.32/3GPP TS 24.008.
+ */
+static int get_ptw_multiplier(float *ptw_multiplier)
+{
+	int err;
+	enum lte_lc_system_mode sys_mode;
+
+	if (ptw_multiplier == NULL) {
+		return -EINVAL;
+	}
+
+	err = lte_lc_system_mode_get(&sys_mode);
+	if (err) {
+		LOG_ERR("Failed to get system mode, error: %d", err);
+		return err;
+	}
+
+	switch (sys_mode) {
+	case LTE_LC_SYSTEM_MODE_LTEM: /* Fall through */
+	case LTE_LC_SYSTEM_MODE_LTEM_GPS:
+		*ptw_multiplier = 1.28;
+		break;
+	case LTE_LC_SYSTEM_MODE_NBIOT: /* Fall through */
+	case LTE_LC_SYSTEM_MODE_NBIOT_GPS:
+		*ptw_multiplier = 2.56;
+		break;
+	case LTE_LC_SYSTEM_MODE_GPS: /* Fall through */
+	case LTE_LC_SYSTEM_MODE_NONE: /* Fall through */
+	default:
+		LOG_ERR("No LTE connection available in this system mode");
+		return -ENOTCONN;
+	}
+
+	return err;
```

Comment:
You should either use the error variable or return directly. Do not use both.
For instance, just return `0` here since you already checked all errors.